content/components/tooltip.mdx

```diff
@@ -3,42 +3,239 @@ title: Tooltip
+- Keep the tooltip text minimal.
+- **Never** include tooltips on non-interactive components (`div`, `span`, `p`).
 
-<Note variant="danger">
-  <Text sx={{display: 'block', fontWeight: 'bold', mb: 2}}>⚠️ Usage Warning ⚠️</Text>
-  <Text>Tooltips as a UI pattern should be our last resort for conveying information because it is hidden by default and often with zero or little visual indicator of its existence.</Text>
-</Note>
+### Best practices
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="456"
+      alt="Icon buttons with tooltips for edit, favorite, and sponsor"
+      src="https://github.com/primer/primitives/assets/18661030/e9f86af8-3792-4794-af68-ad0efccd8bc8"
+    />
+    <Caption>Use a tooltip to give a text label to an icon button</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="456"
+      alt="Non-interactive text element activating a long and wordy tooltip on hover"
+      src="https://github.com/primer/primitives/assets/18661030/1d8d86a2-8ee7-40d2-8934-c3d36f34d9b3"
+    />
+    <Caption>Don't use tooltips on non-interactive elements</Caption>
+  </Dont>
+</DoDontContainer>
+
+<DoDontContainer>
+  <Do>
+    <img
+      width="800"
+      alt=""
+      src="https://github.com/primer/primitives/assets/18661030/d5528bb1-cf2e-41f6-b029-d861fb28375b"
+    />
+    <Caption>Do use tooltips for icon buttons that serve unique purposes</Caption>
+  </Do>
+  <Dont>
+    <img
+      width="800"
+      alt=""
+      src="https://github.com/primer/primitives/assets/18661030/9713ad8b-9f52-44fd-8b46-06d5151b813f"
+    />
+    <Caption>
+      Don’t use tooltips for icon buttons that appear in a group of other controls which share the same functionality,
+      such as opening a context menu.
+    </Caption>
+  </Dont>
+</DoDontContainer>
+
+## Options
+
+### Label
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="icon button with tooltip as a label"
+    src="https://github.com/primer/design/assets/18661030/b06d4fc2-d4e6-482d-a9b9-05eb866ce3e1"
+  />
+  <Box as="p" mt="0">
+    For interactive controls with no visible text label, a tooltip can be used to provide one. This label should be
+    short and descriptive just as you would for a button with visible text.
+  </Box>
+</Box>
+
+### Description
+
+<Box
+  mb={3}
+  display="flex"
+  alignItems="flex-start"
+  flexDirection={['column', 'column', 'column', 'column', 'row']}
+  sx={{gap: 4}}
+>
+  <img
+    width="456"
+    alt="icon button with tooltip as a description"
+    src="https://github.com/primer/design/assets/18661030/2d0cd085-2454-406e-8c99-bc8bd3dd20db"
+  />
+  <Box as="p" mt="0">
+    For interactive controls that require additional context, a tooltip can be used to provide it. The `label` is also
+    required, and the description will be announced to a screen reader in addition to the label. Keep the description
```

Comment:
I think we will want to reword this a bit since this seems to suggest one should set both the `label` and `description` prop of the tooltip API which I don't think is the case. Maybe we want to say that when a tooltip supplements a control as a description, we should make sure the control has an accessible name?

```
<button aria-describedby="tooltip-1">Name of button</button>
<tooltip id="tooltip-1">Some supplementary description</tooltip>
```

Reply:
>we should make sure the control has an accessible name?

I agree with this. Since accessible name could be a visible button text or an `aria-label`, we can maybe divide the "Description" section into two subsections?

Description for buttons that has a visible name
```html
<button aria-describedby="tooltip-1">Name of button</button>
<tooltip id="tooltip-1">Some supplementary description</tooltip>
```

Description for buttons that doesn't have a visible name
```html
<iconbutton aria-label="icon-button-label" />
<tooltip id="tooltip-1">Some supplementary description</tooltip>
```